### DISCUSSION TITLE
Documentation and Learning Gaps

### DISCUSSION THREAD

**Discussion by amcgee** [dhis2/notes#206] [Ideas]
I'd like to collect some of the documentation and learning material gaps we've identified, particularly during the June 2020 app development webinar with 3rd party developer participants.

Feel free to comment below with all your additions, I'll start us off here and collect input from the comments below:

- We're missing "getting started" tutorials
- CodeSandbox isn't perfect, but it's really nice to have "zero-setup" environments available, we should link those from our docs!
- UI components lack context at `ui.dhis2.org`
- Missing documentation for production platform app deployment
- Missing documentation for dynamic data queries
- Some references to `ui-core` still remain, in particular on platform.dhis2.nu and ui.dhis2.nu
- Missing documentation of required peer dependencies for `ui` (from `ui-widgets`)

**Comment by Mohammer5** [comment]
I think what would've helped me the most would've been **new developer on-boarding docs**.
If there was a doc that told me what kind of apps we have (e. g. capture, viz & maintenance), the general data flow in dhis2, links for further reading, etc, I could started being productive way quicker and probably would have known a lot more about the system than I know today.

---

This "Get started" page is very much like what I would like to have:
https://machinelearningmastery.com/start-here/#getstarted

**Reply by amcgee** [reply]
Thanks @Mohammer5 !  @deboragaleano and I are working on this 💯 

This was made quite clear as we onboarded @mediremi @KaiVandivier and @deboragaleano - if any of you want to dump other ideas for documentation gaps please add them here!

**Reply by mediremi** [reply]
Information regarding backporting might be useful, e.g.
- Finding out which versions we currently support
- When creating a PR for [dhis2/dhis2-core](https://github.com/dhis2/dhis2-core) there needs to be 1 for `master` and 1 for each supported version
- How to know which apps don't need backporting (https://github.com/dhis2/dhis2-core/blob/master/dhis-2/dhis-web/dhis-web-apps/README.md)
